packages/addon/lua/gelly/mods/blood-mod/init.lua

```diff
@@ -117,50 +155,48 @@ local function sprayBlood(damageType, attacker, position, force, damage)
 	density = density + damageMultiplier * damage
 
 	gellyx.emitters.Sphere({
-		center = position,
+		center = config.FromEntity and
+				 victim:GetPos() or
+				 position,
 		velocity = velocity,
 		radius = config.CubeSize,
 		density = density,
 		randomness = config.Randomness,
 	})
 end
 
+local function changeMaterialTable(material, table) -- temporary solution until ephemeral presets
+	material.Roughness = 			  table.Roughness
+	material.IsSpecularTransmission = table.IsSpecularTransmission
+	material.RefractiveIndex = 		  table.RefractiveIndex
+	material.Absorption = 			  table.Absorption
+	material.DiffuseColor = 		  table.DiffuseColor
+end
+
 hook.Add(
 	"GellyXDamage",
 	"gelly.builtin.blood-mod",
 	function(victim, attacker, position, force, damage, type)
 		if
 			not victim:IsValid() or
-			not victim:IsPlayer() and
+			(not victim:IsPlayer() and
 			not victim:IsNPC() and
-			not victim:IsRagdoll()
+			not victim:IsRagdoll())
 		then
 			return
 		end
 
-		local bloodcolor = 0
-
-		if victim:GetBloodColor() ~= nil then
-			bloodcolor = victim:GetBloodColor()
-		end
-		
-		local oldabsorption = gellyx.presets.getActivePreset().Material.Absorption
+		local bloodColor = victim:GetBloodColor() or BLOOD_COLOR_RED
 		local material = gellyx.presets.getActivePreset().Material
 
-		if bloodcolor == 0 then -- the blood is red >>> ALSO WHEN IT'S RAGDOLL SINCE THEY DON'T HAVE BLOOD COLOR <<<
-			material.Absorption = Vector(0.05, 0.5, 0.5)
-		elseif bloodcolor == 2 or bloodcolor == 4 then  -- the blood is yellow
-			material.Absorption = Vector(0.25, 0.3, 0.5)
-		else 											-- the blood is green
-			material.Absorption = Vector(0.3, 0.25, 0.5)
-		end 
+		changeMaterialTable(material, BLOOD_COLOR_MATERIALS[bloodColor])
 
 		gelly.SetFluidMaterial(material)
 
-		sprayBlood(type, attacker, position, force, damage)
-
-		material.Absorption = oldabsorption
+		sprayBlood(type, victim, attacker, position, force, damage)
 
+		changeMaterialTable(material, BLOOD_COLOR_MATERIALS[BLOOD_COLOR_RED]) -- (this will change any preset's material into a blood material) only until ephemeral presets
```

Comment:
This has the consequence of now making all future particle spawns inherit the red blood material

Was there anything wrong with the previous `oldMaterial` implementation you had? That didn't seem to have this issue

Reply:
So the problem was that for some reason the materials don't seem to want to be set in the `gellyx.SetFluidMaterial()` function unless they are set manually on top of already existing material table. So i did that but think back now i can just do that the other way around too.